### DISCUSSION TITLE
How could I update browser plugin to current version&

### DISCUSSION THREAD

**Discussion by theromorph** [setvisible/DownZemAll#199] [Q&A]
Hello, man.

Thanks for your time spent on this app!

I can't recall correctly, do I need to keep version of browser plugin the same as the main app? If so, how could I update plugin? Neither Chromium nor Firefox wont accept plugins frin release page.

**Comment by setvisible** [comment]
Hi
The web extension hasn't changed since dec 2020, so doesn't matter to upgrade it or not, and that's why the extension version is different from the application version. 
FYI today I started the process of upgrading in the Mozilla Addons Store, let's see what will happen. With their new policy, I don't know if mozilla still allows this web extension...
Cheers

**Comment by setvisible** [comment]
BtW a web extension can directly be added to a browser.
Firefox and Chrome both propose a sandbox mode for extension development.

**Comment by engelhro** [comment]
It's still confusing to have a version `2.2.0` in the Chrome web store (which works) and simultaneously a version `2.5.5` (currently) available for download…

Especially as this "newer" version doesn't work, at least in Vivaldi (it does not find my DownZemAll installation) while the "older" version from the Chrome web store does and works as expected. Apparently there actually *are* some differences between both versions, despite the sentence above:

> web extension hasn't changed since dec 2020, so doesn't matter to upgrade it or not

@setvisible: could you please clarify or try to remedy that situation? An "up to date" version in the Chrome web store would be fine, but it should be working then :wink:. Please let me know if I can help in debugging the version `2.5.5` issue with the directly installed extension.

**Comment by setvisible** [comment]
Hi,

The chromium extensions ("DownRightNow_chromium_v2.5.5.zip" and "DownRightNow_chromium_v2.2.0.zip") are functionally the same (=same code in both). The difference come maybe from the translations ("messages.json"), since the translations are regularly updated (this is the biggest effort for the last months...).

v2.5.5 is not available yet on the chrome web store, because Google don't let me do that... apparently new policies, and new security criteria prevent it. Every time I submit the extension, there's a Google "peer" review that takes days to come. I don't have time and patience for this.

If you don't mind the translations, v2.2.0 is fine to work with the v2.5.5 application.

Tested successfully with Chrome 97.0 and Firefox 95.0.2 /Windows on my side. Not tested with Vivaldi. If you want to check, please do and feed back any isssue.

PS: I recently added a branch "bugfix-upgrade-web-extension-to-manifest-v3". It's a preliminary effort to convert the web extension (both chromium and firefox versions) to manifest-v3 (the current extension is written for manifest-v2). Didn't work yet. It's a complicated task, because manifest-v3 changes all the logics of events... and the manifest-v3 has some bugs apparently too. If you want to take this task, please do. My personal strategy on this matter is to wait and see, especially how the firefox implementation of manifest-v3 will evolve.

This could maybe help to pass the submission's review ? Moreover I don't know if there are projects that encountered the same issues and if manifest-v3 would helped. It could be interesting to investigate before?

Regards,

**Comment by engelhro** [comment]
> The chromium extensions ("DownRightNow_chromium_v2.5.5.zip" and "DownRightNow_chromium_v2.2.0.zip") are functionally the same (=same code in both). The difference come maybe from the translations ("messages.json"), since the translations are regularly updated (this is the biggest effort for the last months...).

The question then is: why do they behave differently (working vs. non-working, as described)?

> v2.5.5 is not available yet on the chrome web store, because Google don't let me do that... apparently new policies, and new security criteria prevent it. Every time I submit the extension, there's a Google "peer" review that takes days to come. I don't have time and patience for this.

I see. But the information on the [website](https://setvisible.github.io/DownZemAll/category/download.html) is misleading: you state "current version 2.5.5", but the link to the Google web store has then version 2.2.0. Maybe you should just list that version instead?

> Tested successfully with Chrome 97.0 and Firefox 95.0.2 /Windows on my side. Not tested with Vivaldi. If you want to check, please do and feed back any isssue.

Okay, I'll do that, in a separate issue (see #58).

**Comment by PCUser2021** [comment]
> Hi,
> 
> The chromium extensions ("DownRightNow_chromium_v2.5.5.zip" and "DownRightNow_chromium_v2.2.0.zip") are functionally the same (=same code in both). The difference come maybe from the translations ("messages.json"), since the translations are regularly updated (this is the biggest effort for the last months...).
> 
> v2.5.5 is not available yet on the chrome web store, because Google don't let me do that... apparently new policies, and new security criteria prevent it. Every time I submit the extension, there's a Google "peer" review that takes days to come. I don't have time and patience for this.
> 
> If you don't mind the translations, v2.2.0 is fine to work with the v2.5.5 application.
> 
> Tested successfully with Chrome 97.0 and Firefox 95.0.2 /Windows on my side. Not tested with Vivaldi. If you want to check, please do and feed back any isssue.
> 
> PS: I recently added a branch "bugfix-upgrade-web-extension-to-manifest-v3". It's a preliminary effort to convert the web extension (both chromium and firefox versions) to manifest-v3 (the current extension is written for manifest-v2). Didn't work yet. It's a complicated task, because manifest-v3 changes all the logics of events... and the manifest-v3 has some bugs apparently too. If you want to take this task, please do. My personal strategy on this matter is to wait and see, especially how the firefox implementation of manifest-v3 will evolve.
> 
> This could maybe help to pass the submission's review ? Moreover I don't know if there are projects that encountered the same issues and if manifest-v3 would helped. It could be interesting to investigate before?
> 
> Regards,

I can't add your extension manually to Chrome, Can you please help me with how to?
